app/models/project.rb

```diff
@@ -71,6 +71,19 @@ def create_releases_for_branch?(branch)
     release_branch == branch
   end
 
+  # Whether to create a new docker image when the branch is updated.
+  #
+  # branch - The String name of the branch in question.
+  #
+  # Returns true if new releases should be created, false otherwise.
+  def build_docker_image_for_branch?(branch)
+    docker_release_branch == branch
+  end
+
+  def deploy_with_docker?
+    docker_release_branch.present?
```

Comment:
having the branch does not mean it gets deployed with docker ... but that can be another PR ...